src/GafferSceneUI/LightPositionTool.cpp

```diff
@@ -0,0 +1,347 @@
+
+	connectToViewContext();
+	view->contextChangedSignal().connect( boost::bind( &LightPositionTool::connectToViewContext, this  ) );
+
+	view->plugInputChangedSignal().connect( boost::bind( &LightPositionTool::plugInputChanged, this, ::_1 ) );
+
+	selectionChangedSignal().connect( boost::bind( &LightPositionTool::selectionChanged, this, ::_1 ) );
+
+	storeIndexOfNextChild( g_firstPlugIndex );
+}
+
+LightPositionTool::~LightPositionTool()
+{
+}
+
+void LightPositionTool::position( const V3f &shadowPivot, const V3f &shadowPoint )
+{
+	const V3f newCentroid =
+		( shadowPivot - shadowPoint ).normalized() *
+		( m_worldCentroid - shadowPivot ).length() +
+		shadowPivot
+	;
+
+	const M44f centroidTransform = rotationMatrix( m_worldCentroid - shadowPivot, newCentroid - shadowPivot );
+
+	UndoScope undoScope( selection().back().editTarget()->ancestor<ScriptNode>() );
+
+	for( const auto &s : selection() )
+	{
+		// See `RotateTool::buttonPress()` for a description of why we use this relatively
+		// elaborate orientation calculation.
+		Context::Scope scopedContext( s.context() );
+
+		ScenePlug::ScenePath parentPath( s.path() );
+		parentPath.pop_back();
+
+		const M44f worldParentTransform = s.scene()->fullTransform( parentPath );
+		const M44f worldParentTransformInverse = worldParentTransform.inverse();
+		const M44f localTransform = s.scene()->transform( s.path() );
+
+		const M44f worldTransform = s.scene()->fullTransform( s.path() );
+
+		const V3f offset = (
+			( ( worldTransform.translation() - shadowPivot ) * centroidTransform ) + shadowPivot -
+			worldParentTransform.translation() - localTransform.translation()
+		);
+
+		V3f currentYAxis;
+		localTransform.multDirMatrix( V3f( 0.f, 1.f, 0.f ), currentYAxis );
+
+		// Point in the pivot-shadowPoint direction, in local space
+		const V3f targetZAxis = ( shadowPoint - shadowPivot ) * worldParentTransformInverse;
+
+		M44f orientationMatrix = rotationMatrixWithUpDir(
+			V3f( 0.f, 0.f, -1.f ), targetZAxis, currentYAxis
+		);
+
+		V3f originalRotation;
+		extractEulerXYZ( localTransform, originalRotation );
+		const M44f originalRotationMatrix = M44f().rotate( originalRotation );
+
+		const M44f relativeMatrix = originalRotationMatrix.inverse() * orientationMatrix;
+
+		V3f relativeRotation;
+		extractEulerXYZ( relativeMatrix, relativeRotation );
+
+		TranslationRotation( s ).apply( offset, relativeRotation );
+	}
+}
+
+bool LightPositionTool::affectsHandles( const Gaffer::Plug *input ) const
+{
+	if( TransformTool::affectsHandles( input ) )
+	{
+		return true;
+	}
+
+	return input == scenePlug()->transformPlug();
+}
+
+void LightPositionTool::updateHandles( float rasterScale )
+{
+
+}
+
+void LightPositionTool::selectionChanged( const TransformTool &tool )
+{
+	if( activePlug()->getValue() )
+	{
+		if( selection().empty() )
+		{
+			return;
+		}
+
+		m_worldCentroid = V3f( 0 ) * selection().back().orientedTransform( Orientation::World );
```

Comment:
I've been getting a few of these errors in my terminal :

```ERROR : Emitting signal : Selection is not editable```

Not 100% sure it's coming from here, but `orientedTransform()` is one of the functions that could throw that exception (in TransformTool.cpp).